cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -57,7 +57,15 @@ cdef class Buffer:
         return self
 
     def __del__(self):
-        self.close()
+        self.safe_close()
+
+    cpdef safe_close(self, stream: Stream = None, is_shutting_down=is_shutting_down):
+        if self._ptr and self._mr is not None:
+            if not is_shutting_down():
+                self._mr.deallocate(self._ptr, self._size, stream)
+            self._ptr = 0
+            self._mr = None
+            self._ptr_obj = None
```

Comment:
Why is setting these to `None`/0 required? Doesn't this happen automatically as part of either GC, the usual refcount-equals-zero behavior, or interpreter shutdown? You're only calling `safe_close` from `__del__` so nullifying these fields seems like it might be superfluous.